File: Java/hellointerview/02-sliding-window/maxSubArraySum.java
public class Solution {
  public int maxSubArraySum(int[] nums, int k) {
    int result = 0;
    int start = 0;
    int windowSum = 0;

    for (int end = 0; end < nums.length; end++) {
      windowSum += nums[end];

      if (end - start + 1 == k) {
        // if window size equals target size
        result = Math.max(result, windowSum);

        // now prepare for next iteration
        windowSum -= nums[start];
        start++;
      }
    }

    return result;
  }
}

File: TS/hellointerview/sliding-window/maxSubarraySum.ts
function maxSubarraySum(nums: number[], k: number): number {
  let start = 0;
  let windowSum = 0;
  let maxSum = 0;

  for (let end = 0; end < nums.length; end++) {
    windowSum += nums[end];

    if (end - start + 1 === k) {
      maxSum = Math.max(maxSum, windowSum);
      windowSum -= nums[start];
      start++;
    }
  }

  return maxSum;
}
